core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerClientProvider.scala

```diff
@@ -0,0 +1,39 @@
+package org.broadinstitute.dsde.rawls.billing
+
+import bio.terra.profile.api.{AzureApi, ProfileApi}
+import bio.terra.profile.client.ApiClient
+
+/**
+ * Implementors of this trait know how to instantiate billing profile manager client
+ * classes for use with the billing profile manager service.
+ */
+trait BillingProfileManagerClientProvider {
+  def getApiClient(accessToken: String): ApiClient
+
+  def getAzureApi(accessToken: String): AzureApi
```

Comment:
My first thought when reading this… why is there an Azure Api and not a GCP Api? What exactly is the ApiClient and the ProfileApi… is one of those really GCP-specific? Perhaps some documentation would help me understand the role of these 3 client classes, but for now I will continue reading and see if it all becomes clear.

Reply:
Also, it appears that `getApiClient` is only used by the other 2 methods. So does it need to be exposed as a public part of the interface?

Reply:
Having gotten further, I am now reminded that BPM does not support GCP yet. So presumably someday there may be a `getGcpApi` method?

Reply:
Yes, this is a structure dictated by the auto generated billing profile manager client JAR. These `AzureApi`, `FooApi` types directly correspond to the controllers/sections on the BPM swagger (https://bpm.dsde-dev.broadinstitute.org/swagger-ui.html).